apps/prairielearn/src/tests/courseElementExtension.test.js

```diff
@@ -76,7 +76,10 @@ describe('Course element extensions', function () {
   });
 
   describe('Extensions can insert client-side assets into the page', function () {
-    before('set up testing server', helperServer.before(EXAMPLE_COURSE_PATH));
+    before(
+      'set up testing server',
+      helperServer.before(EXAMPLE_COURSE_PATH, { allowSyncFailure: true })
```

Comment:
I don't love this. I like the fact that tests would previously actually fail if there was a sync error/warning, which would indicate that there's something wrong with the example course.

Can you elaborate on why sync failures would actually occur? If they occur during tests, that means they're also likely to occur in people's PrairieLearn installations when syncing the example course, which is undesirable.

Reply:
Ya, so the problem here was that I want to have an example assessment in the example course that consumes a question from the test course, but the initial sync was failing because at the point of initial sync, the sharing sets were not set up yet. 

Now though, I have a config option that makes it so question imports are not validated. If we leave that off for the first part of the tests, and leave that off for local dev, it should allow for sync success in those places, but we can make sure that config option is always on in production. 

Does that sound all right?

Reply:
fixed